packages/cactus-plugin-bungee-hermes/src/main/typescript/strategy/strategy-ethereum.ts

```diff
@@ -0,0 +1,223 @@
+    const config = new Configuration({
+      basePath: networkDetails.connectorApiPath,
+    });
+    const ethereumApi = new EthereumApi(config);
+
+    const ledgerStates = new Map<string, State>();
+    const assetsKey =
+      stateIds.length == 0
+        ? await this.getAllAssetsKey(networkDetails, ethereumApi)
+        : stateIds;
+    this.log.debug("Current assets detected to capture: " + assetsKey);
+    for (const assetKey of assetsKey) {
+      const { transactions, values, blocks } = await this.getAllInfoByKey(
+        assetKey,
+        networkDetails,
+        ethereumApi,
+      );
+
+      const state = new State(assetKey, values, transactions);
+
+      const stateProof = new StateProof(
+        state.getValue(),
+        parseInt(state.getVersion()),
+        state.getId(),
+      );
+      const blocksHash: string[] = [];
+      for (const block of blocks.values()) {
+        if (blocksHash.indexOf(block.hash as string) !== -1) {
+          continue;
+        }
+        blocksHash.push(block.hash as string);
+        stateProof.addBlock({
+          blockHash: block.hash as string,
+          blockCreator: block.miner as string,
+          blockSigners: [], // Non applicable for ethereum
+        });
+      }
+      state.setStateProof([stateProof]);
+      ledgerStates.set(assetKey, state);
+    }
+    return ledgerStates;
+  }
+
+  async getAllAssetsKey(
+    networkDetails: EthereumNetworkDetails,
+    api: EthereumApi,
+  ): Promise<string[]> {
+    const parameters = {
+      contract: {
+        contractName: networkDetails.contractName,
+        keychainId: networkDetails.keychainId,
+      },
+      invocationType: EthContractInvocationType.Call,
+      methodName: "getAllAssetsIDs",
+      params: [],
+      signingCredential: networkDetails.signingCredential,
+    };
+    const response = await api.invokeContractV1(
+      parameters as InvokeContractV1Request,
+    );
+    if (response != undefined && response.data.callOutput != undefined) {
+      return response.data.callOutput;
+    }
+
+    throw Error(
+      "getAllAssetsIDs contract function invocation failed or response is invalid",
+    );
+  }
+
+  async getAllInfoByKey(
+    key: string,
+    networkDetails: EthereumNetworkDetails,
+    api: EthereumApi,
+  ): Promise<{
+    transactions: Transaction[];
+    values: string[];
+    blocks: Map<string, EvmBlock>;
+  }> {
+    const filter = {
+      fromBlock: "earliest",
+      toBlock: "latest",
+      address: networkDetails.contractAddress,
+      topics: [null, Web3.utils.keccak256(key)], //filter logs by asset key
+    };
+    const getLogsReq: InvokeRawWeb3EthMethodV1Request = {
+      methodName: "getPastLogs",
+      params: [filter],
+    };
+    const response = await api.invokeWeb3EthMethodV1(getLogsReq);
+    if (
+      response == undefined ||
+      !(response.status >= 200 && response.status <= 299)
+    ) {
+      throw Error("Error on getting past logs");
```

Comment:
@eduv09 Looking a little deeper at the error handling: Please include the root cause here as well. `RuntimeError` can have a `cause` property which can be another `Error` or a `string`. The way you throw this error completely hides and swallows the original problem and will make people who have to debug this in production be disappointed when all they see in the error log is `Error on getting past logs` and literally nothing else. Ideally you'd break it down to multiple different checks like an if condition for the response being falsy (use the ! operator to check for falsyness instead of `== undefined` ) and another if condition for checking if the status code is in the valid range. Then you can throw different errors with d ifferent messages that are way more informative that way. 
Please also make sure to include the status code if it wasn't in the valid range. There could be valuable information hidden there about the nature of the problem that the code is having. 

For example if the return codes are 4xx then it's user error and you should throw an error accordingly. If it's 5xx, then it's supposedly a bug in our code.

Take a look at the besu connector's deploy contract and how I refactored it to use these principles a while back:

```typescript
public async deployContract(
    req: DeployContractSolidityBytecodeV1Request,
  ): Promise<DeployContractSolidityBytecodeV1Response> {
    const fnTag = `${this.className}#deployContract()`;
    Checks.truthy(req, `${fnTag} req`);
    if (isWeb3SigningCredentialNone(req.web3SigningCredential)) {
      throw createHttpError[400](
        `${fnTag} Cannot deploy contract with pre-signed TX`,
      );
    }
    const { keychainId, contractName } = req;
    if (!keychainId || !req.contractName) {
      const errorMessage = `${fnTag} Cannot deploy contract without keychainId and the contractName.`;
      throw createHttpError[400](errorMessage);
    }

    const keychainPlugin = this.pluginRegistry.findOneByKeychainId(keychainId);

    if (!keychainPlugin) {
      const errorMessage =
        `${fnTag} The plugin registry does not contain` +
        ` a keychain plugin for ID:"${req.keychainId}"`;
      throw createHttpError[400](errorMessage);
    }

    if (!keychainPlugin.has(contractName)) {
      const errorMessage =
        `${fnTag} Cannot create an instance of the contract instance because` +
        `the contractName in the request does not exist on the keychain`;
      throw new createHttpError[400](errorMessage);
    }
```

Reply:
Thank you @petermetz ! I changed the error handling in all strategies taking into consideration this comment. Hopefully it is being accurately handled this time.
Let me know any improvements/corrections you see fit.